packages/s3-store/index.ts

```diff
@@ -484,9 +536,11 @@ export class S3Store extends DataStore {
     else if (size <= this.preferredPartSize * this.maxMultipartParts) {
       optimalPartSize = this.preferredPartSize
       // The upload is too big for the preferred size.
-      // We devide the size with the max amount of parts and round it up.
+      // We divide the size with the max amount of parts and round it up.
+    } else if (size % this.maxMultipartParts === 0) {
+      optimalPartSize = size / this.maxMultipartParts
     } else {
-      optimalPartSize = Math.ceil(size / this.maxMultipartParts)
+      optimalPartSize = size / this.maxMultipartParts + 1
```

Comment:
Why was the `Math.ceil` removed? If we reach this conditional block, we know that `size / this.maxMultipartParts` is not an integer, so rounding it would be sense to get rid of the fractional part.

Reply:
I based this change on this:
https://github.com/tus/tusd/blob/main/pkg/s3store/s3store.go#L1176

There is no rounding as far as i can see

Reply:
Go automatically does rounding :)

Reply:
To add some more explanation to @Murderlon's comment: The code from tusd uses integers. Integer division is different than floating point division in that the fractional is ignored. For example:

```
14 / 5 == 2 // integer division
14.0 / 5 == 2.8 // float divison
```

That's why you want `Math.floor` when trying to replicate the Go logic, or `Math.ceil` when you want to replicate the `integer division + 1` from tusd.

Reply:
Ah yes, very good point! I totally missed that the data types were ints and not float. Will bring the rounding back :)